mungegithub/mungers/matchers/comment/interactions.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package comment
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/google/go-github/github"
+)
+
+// BotMessage identifies message sent by the bot, with the following format:
+// [TYPEOFMESSAGE] Optional arguments
+type BotMessage string
+
+// Match returns true if the comment is a message from the bot
+func (b BotMessage) Match(comment *github.IssueComment) bool {
+	if comment.Body == nil {
+		return false
+	}
+	match, _ := regexp.MatchString(fmt.Sprintf("^\\[%s\\]", string(b)), *comment.Body)
```

Comment:
You can quote with backticks to prevent any doubt about whether \ is
interpreted specially.

On Mon, Aug 15, 2016 at 7:42 PM, Antoine Pelisse <notifications@github.com>
wrote:

> In mungegithub/mungers/matchers/comment/interactions.go
> <https://github.com/kubernetes/contrib/pull/1530#discussion_r74868921>:
>
> > +	"fmt"
> > +	"regexp"
> > +
> > +	"github.com/google/go-github/github"
> > +)
> > +
> > +// BotMessage identifies message sent by the bot, with the following format:
> > +// [TYPEOFMESSAGE] Optional arguments
> > +type BotMessage string
> > +
> > +// Match returns true if the comment is a message from the bot
> > +func (b BotMessage) Match(comment *github.IssueComment) bool {
> > +	if comment.Body == nil {
> > +		return false
> > +	}
> > +	match, _ := regexp.MatchString(fmt.Sprintf("^\\[%s\\]", string(b)), *comment.Body)
>
> No, The double slash is converted by Go into single slash. This final
> single slash is used to explicitly ask [ to not count as a group, but to
> match the actual character. The goal is to match (it should be described
> above in the comment): [something]
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/contrib/pull/1530/files/fc41d11819c058ca5fafa87d762b8f1867dc898b#r74868921>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAngljxxx3MurcXo-DTMQnq-7HNdfHb7ks5qgSONgaJpZM4JjXFP>
> .
>